### ISSUE TITLE
vmalert-tool: inconsistent with vmalert on how to treat external labels

### ISSUE THREAD

**Issue by umezawatakeshi** [VictoriaMetrics/VictoriaMetrics#6735]
### Describe the bug

`vmalert-tool unittest` is inconsistent with `vmalert` on how to treat `$externalLabels` in alert templates. `vmalert` sets labels specified by `-external.label` options to `$externalLabels` but `vmalert-tool unittest` does not set labels specified by `external_labels` in test files to `$externalLabels`.

Both of them add those labels to `$labels`. (consistent behavior)

### To Reproduce

Suppose the following alert rule:
```yaml
groups:
- name: testrule
  rules:
  - alert: TestAlert
    annotations:
      summary: '$labels.foo is "{{ $labels.foo }}" and $externalLabels.foo is "{{ $externalLabels.foo }}"'
    expr: |
      1
    for: 1m
```
`vmalert -external.label=foo=bar` will generate an alert like:
```
$ curl -s http://localhost:8080/api/v1/alerts | jq .
{
  "status": "success",
  "data": {
    "alerts": [
(snip)
      {
        "state": "firing",
        "name": "TestAlert",
        "value": "1",
        "labels": {
          "alertgroup": "testrule",
          "alertname": "TestAlert",
          "foo": "bar",
        },
        "annotations": {
          "summary": "$labels.foo is \"bar\" and $externalLabels.foo is \"bar\""
        },
        "activeAt": "2024-07-30T05:24:30Z",
        "id": "10621654793665610911",
        "rule_id": "12340062498589138746",
        "group_id": "14590765261014202938",
        "expression": "1\n",
        "source": "http://vmalert-vmalert-largeset-76878f7579-89kdk:8080/vmalert/alert?group_id=14590765261014202938&alert_id=10621654793665610911",
        "restored": false,
        "stabilizing": false
      },
(snip)
    ]
  }
}
```
However, with the following test file,
```yaml
rule_files:
   - test_externallabel_rule.yaml

tests:
  - interval: 1m
    external_labels:
      foo: bar
    alert_rule_test:
      - eval_time: 1m
        alertname: TestAlert
        exp_alerts:
          - exp_labels:
              foo: bar
            exp_annotations:
              summary: '$labels.foo is "bar" and $externalLabels.foo is "bar"'
```
`vmalert-tool unittest` will raise the error:
```
$ ./vmalert-tool unittest --disableAlertgroupLabel --files test_externallabel_config.yaml

Unit Testing:  test_externallabel_config.yaml
evaluation_interval set to 1m by default
  FAILED

failed to run unit test for file "rule_files:\n   - test_externallabel_rule.yaml\n\ntests:\n  - interval: 1m\n    external_labels:\n      foo: bar\n    alert_rule_test:\n      - eval_time: 1m\n        alertname: TestAlert\n        exp_alerts:\n          - exp_labels:\n              foo: bar\n            exp_annotations:\n              summary: '$labels.foo is \"bar\" and $externalLabels.foo is \"bar\"'\n":
[
    groupname: , alertname: TestAlert, time: 1m0s,
        exp:[
            0:
              Labels:{alertname="TestAlert", foo="bar"}
              Annotations:{summary="$labels.foo is \"bar\" and $externalLabels.foo is \"bar\""}
            ],
        got:[
            0:
              Labels:{alertname="TestAlert", foo="bar"}
              Annotations:{summary="$labels.foo is \"bar\" and $externalLabels.foo is \"\""}
            ] ]2024/07/31 19:21:27 unittest failed
```

### Version

Both of the following
- `vmalert-tool version vmalert-tool-20240411-145824-tags-v1.100.1-0-g9ee51e34c`
- `vmalert-tool version vmalert-tool-20240717-185205-tags-v1.102.0-0-g65bb429b81`

### Additional information

Just FYI: `promtool test rules` adds those labels to `$externalLabels` but not to `$labels`. (different from both `vmalert` and current `vmalert-tool`!)

**Comment by Haleygo** [comment]
Thanks for the detailed report!
The `external_labels` options in the [test file](https://docs.victoriametrics.com/vmalert-tool/#test-file-format) is not designed to act the same like vmalert's `-external.label` flag, but only for adding extra labels, so it can't be used for [templating](https://docs.victoriametrics.com/vmalert/#templating) rule's labels or annotations.
But as you mentioned, this causes inconsistency between vmalert and vmalert-tool. Therefore, I think the appropriate fix should be deprecate the `external_labels` options and add `-external.label` flag for vmalert-tool as well.

**Comment by hagen1778** [comment]
https://github.com/VictoriaMetrics/VictoriaMetrics/issues/6735#issuecomment-2271463425 was addressed in https://github.com/VictoriaMetrics/VictoriaMetrics/pull/6766 and will be included into the next release.

**Comment by hagen1778** [comment]
This change was included into [v1.103.0](https://docs.victoriametrics.com/changelog/#v11030).
